### PR TITLE
Properly validate files when multiple files are used

### LINKED ISSUE #2203: Stack trace when an override file is empty
If I create an empty `docker-compose.override.yml` in a project, `docker-compose up` throws the following error:

```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 53, in main
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 23, in sys_dispatch
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 26, in dispatch
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 163, in perform_command
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 54, in project_from_options
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 90, in get_project
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.config.config", line 215, in load
  File "/compose/build/docker-compose/out00-PYZ.pyz/compose.config.config", line 205, in merge_services
TypeError: 'NoneType' object is not iterable
```

We should instead show the same `Top level object needs to be a dictionary` error that you get if your `docker-compose.yml` file is empty.

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #2203 

Removed the single-use decorators so the functionality can be used directly as a function.

### CODE REVIEW

File: compose/config/config.py [hunk 1]
```diff
@@ -165,13 +164,11 @@ def find_candidates_in_parent_dirs(filenames, path):
     return (candidates, path)
 
 
-@validate_top_level_object
-@validate_service_names
```
Comment (mnowster) [inline compose/config/config.py]: You've removed `validate_service_names` without putting it back in elsewhere. Was this on purpose? You still need this validation to happen, validating a `top_level_object` as a dict on it's own isn't enough.

File: compose/config/config.py [hunk 2]
```diff
@@ -165,13 +164,11 @@ def find_candidates_in_parent_dirs(filenames, path):
...
 def pre_process_config(config):
-    """
-    Pre validation checks and processing of the config file to interpolate env
+    """Pre validation checks and processing of the config file to interpolate env
     vars returning a config dict ready to be tested against the schema.
     """
+    validate_top_level_object(config)
```
Comment (mnowster) [inline compose/config/config.py]: Given the direction you're going with this approach, I don't think it makes sense to have this being called as part of `pre_process_config`. As you've already called `validate_top_level_object` for each config file, so there's no need to include it in this function and call it again.

File: compose/config/config.py (hunk 1 above)
Comment (dnephin) [inline compose/config/config.py]: I moved it into `validate_top_level_object()` so they're always called together now.

File: compose/config/config.py (hunk 2 above)
Comment (dnephin) [inline compose/config/config.py]: Ya, i was going back and forth on this one. This is also called from the `extends` resolving.  I was considering just dropping this method and calling both functions. I think I'll do that.

File: compose/config/config.py (hunk 1 above)
Comment (mnowster) [inline compose/config/config.py]: dammit, sorry, I missed that.

File: compose/config/config.py (hunk 2 above)
Comment (mnowster) [inline compose/config/config.py]: Yeh, I think given how the code has changed that it doesn't fit into the model of how things work now and the different places it needs to be called. So I'm +1 with it being dropped.

**Comment by dnephin** [comment]
Removed `pre_process_config()`

**Comment by mnowster** [comment]
BOOM. 

LGTM>

**Pull request merged**